src/global_scheduler/global_scheduler_algorithm.c

```diff
@@ -0,0 +1,33 @@
+#include "task.h"
+#include "task_table.h"
+
+#include "global_scheduler_algorithm.h"
+
+void handle_task_waiting(global_scheduler_state *state, task *original_task) {
```

Comment:
I'm assuming that this code is throwaway, but can we document what the current algorithm is anyway?